autoload/bazel/workspace.vim

```diff
@@ -0,0 +1,51 @@
+""
+" @section Introduction, intro
+" @library
+" Implements common path functionality used by bazel-related plugins.  For
+" more details about Build file structure/terminology, see:
+" https://bazel.build/versions/master/docs/build-ref.html
+
+""
+" @private
+" Returns the root path for a Bazel workspace file (given the current working
+" directory). {path} is the absolute path.
+function! bazel#workspace#GetBazelRootForCwd() abort
+  return bazel#workspace#GetBazelRootPath(getcwd())
+endfunction
+
+
+""
+" @private
+" Returns the root path for Bazel workspace. {path} is the absolute path.
+"
+" I.e., if you are in directory: /Users/$USER/myproject/foo/bar/biff
+" And there is a WORKSPACE file at: /Users/$USER/myproject/WORKSPACE
+" Return /home/$USER/myproject. If no workspace can be found, return ''
```

Comment:
Could this throw a NotFound error instead of returning `''`? A caller could always catch and ignore errors if desired.

Reply:
I could; this logic came from internal code, so I'm inclined to leave it.. It's not clear to me that throwing an exception is better; having written so much go code recently, I would towards returning a raw type because it leads to explicit handling.

Reply:
In go the explicit handling is usually nil or some special type that's not likely to accidentally be used as a real value, right? My concern here is that dummy paths would in a lot of cases seem to work but do the wrong thing (and vimmers IME pretty much never handle corner cases until they blow up in practice). And IIRC the pre-existing code just inherited that from vim's path handling helpers before we started noticing what gnarly bugs we got thanks to that behavior, for instance fnamemodify silently expanding it to CWD.

Reply:
Done